### PR TITLE
Project edits for pr517

### PR DESCRIPTION
**Pull request by vir-mir**
- Modification in tests
- Removing old style file examples
- Removing old style (yield from) in docs
- Rename psycopg2 to psycopg2-binary in docs
- added note in docs about going to python3.5.2+
- Interim fix due to issues #533 
- refactor docs style
- aded the example code 

@asvetlov @jettify since the changes are not compatible with previous versions, I propose to upgrade to major version `1.0.0`

### CODE REVIEW

**Comment by gjcarneiro** [comment]
In the next release notes, besides saying it requires async/await, there should probably be a note about dropping of the deprecated non-asynchronous iterator protocol (`__iter__`).  Now only `__aiter__` is available.

So this works:

```python
async for row in cursor: ...
```

This used to work, but was already deprecated, and no longer works:

```python
for row in cursor: ...
```

**Comment by vir-mir** [comment]
@gjcarneiro thx, this is a good point!

File: .readthedocs.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: 2
+
+build:
+  image: latest
+
+python:
+  version: 3.7
```
Comment (webknjaz) [inline .readthedocs.yml]: AFAIR they don't ship 3.7 in their images currently.

File: .readthedocs.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  install:
+      - requirements: requirements.txt
+      - method: pip
+        path: .
+   system_packages: true
```
Comment (webknjaz) [inline .readthedocs.yml]: This looks like a broken indentation

File: .travis.yml [hunk 3]
```diff
@@ -11,10 +11,10 @@ python:
 - "nightly"
 
 .mixtures:
-- &py3_7-base
-  python: "3.7"
-  dist: xenial
-  sudo: required
+  - &py3_7-base
```
Comment (webknjaz) [inline .travis.yml]: please move it back. extra indentation is not needed

File: .travis.yml [hunk 4]
```diff
@@ -39,12 +39,16 @@ env:
...
+  allow_failures:
+    - python: "nightly"
+
 cache:
   directories:
   - $HOME/.cache/pip
 
 before_cache:
-- rm -f $HOME/.cache/pip/log/debug.log
+  - rm -f $HOME/.cache/pip/log/debug.log
```
Comment (webknjaz) [inline .travis.yml]: not related. it's hard to review style changes combined with something actually useful which just gets lost among unrelated changes.

File: .travis.yml [hunk 5]
```diff
@@ -39,12 +39,16 @@ env:
   - PYTHONASYNCIODEBUG=x
   - PYTHONASYNCIODEBUG=
 
+matrix:
+  allow_failures:
+    - python: "nightly"
```
Comment (webknjaz) [inline .travis.yml]: the established indentation style across this file doesn't require extra space in sequences.

File: .travis.yml (hunk 4 above)
Comment (vir-mir) [reply .travis.yml]: ok. I will correct

File: .travis.yml (hunk 3 above)
Comment (vir-mir) [reply .travis.yml]: ok. I will correct

File: .readthedocs.yml (hunk 2 above)
Comment (vir-mir) [reply .readthedocs.yml]: ok. I will correct

File: .readthedocs.yml (hunk 1 above)
Comment (vir-mir) [reply .readthedocs.yml]: yes. now yes. maybe it was a crash in travis-ci
but python version 3.8.0a0 https://travis-ci.com/aio-libs/aiohttp/jobs/173498419
configuration of this build  https://travis-ci.com/aio-libs/aiohttp/jobs/173498419/config

the python3.7 version is stable, then it is worth pointing it out explicit. And build `nightly` is worth pointing with probability of error.

**Comment by vir-mir** [comment]
@asvetlov @popravich  You are noted in the documentation support here https://readthedocs.org/projects/aiopg/ 

The question is, are the hooks configured to build the master and the tags? I watch the last build was a long time ago. https://readthedocs.org/projects/aiopg/builds/

**Comment by asvetlov** [comment]
Thanks for the PR.
What is your account on readthedocs?
I want to add you as the project admin

File: docs/conf.py [hunk 6]
```diff
@@ -64,14 +68,16 @@ def get_version(release):
...
 
 # The master toctree document.
 master_doc = 'index'
 
 # General information about the project.
 project = 'aiopg'
-copyright = '2014-2016, Andrew Svetlov'
+date = datetime.date.today()
+
+copyright = '2014-{year}, Andrew Svetlov'.format(year=date.year)
```
Comment (asvetlov) [inline docs/conf.py]: @vir-mir I think you can add your name here (and and in all other places where it makes sense)

File: docs/index.rst [hunk 7]
```diff
@@ -71,23 +94,13 @@ please go to psycopg docs: http://initd.org/psycopg/docs/
 .. note::
 
    Throughout this documentation, examples utilize the `async/await` syntax
-   introduced by :pep:`492` that is only valid for Python 3.5+.
+   introduced by :pep:`492` that is only valid for Python 3.5.2+.
```
Comment (asvetlov) [inline docs/index.rst]: I think you can just drop the note.
I made a sense 4 years ago, but now nobody uses `yield from` for asyncio in their code.

**Comment by asvetlov** [comment]
I've configured readthedocs hooks just now.

File: .readthedocs.yml (hunk 1 above)
Comment (webknjaz) [reply .readthedocs.yml]: @vir-mir this is not a Travis config, it's a RTD config

File: Makefile [hunk 8]
```diff
@@ -24,6 +35,9 @@ cov cover coverage: flake
 cov-ci: flake
 	py.test -svvv -rs --cov --cov-report=term tests --pg_tag all
 
+clean-pip:
+	pip freeze | grep -v "^-e" | xargs pip uninstall -y
```
Comment (webknjaz) [inline Makefile]: maybe `pip list`?

File: aiopg/cursor.py [hunk 9]
```diff
@@ -40,7 +39,7 @@ def description(self):
         6.  null_ok: always None as not easy to retrieve from the libpq.
 
         This attribute will be None for operations that do not
-        return rows or if the cursor has not had an operation invoked
+        anreturn rows or if the cursor has not had  operation invoked
```
Comment (brabadu) [inline aiopg/cursor.py]: Looks like a side effect of using multiple cursors in text editor

```suggestion
        return rows or if the cursor has not had an operation invoked
```

File: docs/one_cursor.rst [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+but the :ref:`aiopg-core-cursor` remained open.
+
+Implementation
+--------------
+
+It was decided to select one connection, one :ref:`aiopg-core-cursor`.
+For the interface :ref:`aiopg-core-connection` interface
+over :term:`psycopg2-binary`,
+we added the :meth:`aiopg.Connection.free_cursor`
+method to clean the course if it is open.
```
Comment (brabadu) [inline docs/one_cursor.rst]:
```suggestion
method to clean the cursor if it is open.
```

File: .readthedocs.yml [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+build:
+  image: latest
+
+python:
+  version: 3.7
+  install:
+  - requirements: requirements.txt
+  - method: pip
+  path: .
```
Comment (webknjaz) [inline .readthedocs.yml]: This is underindented.

Comment (webknjaz) [reply .readthedocs.yml]:
```suggestion
    path: .
```

File: CHANGES.txt [hunk 12]
```diff
@@ -1,119 +1,116 @@
-CHANGES
--------
-
 0.16.0 (2019-01-25)
 ^^^^^^^^^^^^^^^^^^^
 
-* Fix select priority name (#525)
+* Fix select priority name `#525 <https://github.com/aio-libs/aiopg/issues/525>`_
```
Comment (webknjaz) [inline CHANGES.txt]: Maybe add extlinks for this?
https://github.com/cherrypy/cheroot/blob/92490dd/docs/conf.py#L34-L40

File: CONTRIBUTING.rst [hunk 13]
```diff
@@ -4,29 +4,37 @@ Instruction for contributors
 Developer environment
 ---------------------
 
-First clone the git repo::
+First clone the git repo
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
First clone the git repo:
```

File: CONTRIBUTING.rst [hunk 14]
```diff
@@ -4,29 +4,37 @@ Instruction for contributors
...
-   $ cd aiopg
+.. code-block:: shell
+
+    $ git clone git@github.com:aio-libs/aiopg.git
+    $ cd aiopg
 
 After that you need to create and activate a virtual environment.  I
 recommend using :term:`virtualenvwrapper` but just :term:`virtualenv` or
-:term:`venv` will also work.  For ``virtualenvwrapper``::
+:term:`venv` will also work. For :term:`virtualenvwrapper`
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
:term:`venv` will also work. For :term:`virtualenvwrapper`:
```

File: CONTRIBUTING.rst [hunk 15]
```diff
@@ -4,29 +4,37 @@ Instruction for contributors
...
+
+.. code-block:: shell
+
+    $ mkvirtualenv aiopg -p `which python3`
 
-   $ mkvirtualenv aiopg -p `which python3`
+For `venv` (for example; put the directory wherever you want)
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
For `venv` (for example; put the directory wherever you want):
```

File: CONTRIBUTING.rst [hunk 16]
```diff
@@ -4,29 +4,37 @@ Instruction for contributors
...
-For ``venv`` (for example; put the directory wherever you want)::
+.. code-block:: shell
 
-   $ python3 -m venv ../venv_directory
-   $ source ../venv_directory/bin/activate
+    $ python3 -m venv ../venv_directory
+    $ source ../venv_directory/bin/activate
 
-Just as when doing a normal install, you need the :term:`libpq` library::
+Just as when doing a normal install, you need the :term:`libpq` library
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
Just as when doing a normal install, you need the :term:`libpq` library:
```

File: CONTRIBUTING.rst [hunk 17]
```diff
@@ -4,29 +4,37 @@ Instruction for contributors
...
 
-   $ sudo apt-get install libpq-dev
+.. code-block:: shell
+
+    $ sudo apt-get install libpq-dev
 
 **UPD**
 
-The latest ``aiopg`` test suite uses docker container for running
+The latest `aiopg` test suite uses docker container for running
```
Comment (webknjaz) [inline CONTRIBUTING.rst]: In RST single backticks produce `italic text` while double backticks make ``inline code``.

```suggestion
The latest ``aiopg`` test suite uses docker container for running
```

File: CONTRIBUTING.rst [hunk 18]
```diff
@@ -35,37 +43,47 @@ No local Postgres server needed.
...
 
-   $ pip install -Ue .
-   $ pip install -Ur requirements.txt
+    $ pip install -Ue .
+    $ pip install -Ur requirements.txt
 
 That's all.
 
-To run all of the *aiopg* tests do::
+To run all of the *aiopg* tests do
```
Comment (webknjaz) [inline CONTRIBUTING.rst]: You still want a single colon here:
```suggestion
To run all of the *aiopg* tests do:
```
Double colon creates a block and is rendered as a single colon, single colon just does nothing and is rendered as is.

File: CONTRIBUTING.rst [hunk 19]
```diff
@@ -35,37 +43,47 @@ No local Postgres server needed.
...
+
+    $ make test
 
 This command runs :term:`pep8` and :term:`pyflakes` first and then executes
 the *aiopg* unit tests.
 
 
 When you are working on solving an issue you will probably want to run
-some specific test, not the whole suite::
+some specific test, not the whole suite
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
some specific test, not the whole suite:
```

File: CONTRIBUTING.rst [hunk 20]
```diff
@@ -35,37 +43,47 @@ No local Postgres server needed.
...
+
+.. code-block:: shell
 
-   $ py.test -s -k test_initial_empty
+    $ py.test -s -k test_initial_empty
 
 For debug sessions I prefer to use :term:`ipdb`, which is installed
-as part of the development tools.  Insert the following line into your
```
Comment (webknjaz) [inline CONTRIBUTING.rst]: Some style guides (like PEP 257) actually require double space after dots. But it's up to you here...

File: CONTRIBUTING.rst [hunk 21]
```diff
@@ -35,37 +43,47 @@ No local Postgres server needed.
...
+as part of the development tools. Insert the following line into your
 code in the place where you want to start interactively debugging the
-execution process::
+execution process
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
execution process:
```

File: CONTRIBUTING.rst [hunk 22]
```diff
@@ -35,37 +43,47 @@ No local Postgres server needed.
...
 
-   import ipdb; ipdb.set_trace()
+.. code-block:: py3
+
+    import ipdb; ipdb.set_trace()
 
 The library is reasonably well covered by tests.  There is a make
-target for generating the coverage report::
+target for generating the coverage report
```
Comment (webknjaz) [inline CONTRIBUTING.rst]:
```suggestion
target for generating the coverage report:
```

File: Makefile [hunk 23]
```diff
@@ -1,15 +1,26 @@
 # Some simple testing tasks (sorry, UNIX only).
 
 doc:
-	cd docs && make html
+	cd docs && rm -rf _build/html && make html
```
Comment (webknjaz) [inline Makefile]: Better create some `clean-docs:` target and depend on it...

File: aiopg/__init__.py [hunk 24]
```diff
@@ -28,26 +29,30 @@
...
                   'a': 'alpha',
                   'b': 'beta',
                   None: 'final'}
         releaselevel = levels[match.group('releaselevel')]
         serial = int(match.group('serial')) if match.group('serial') else 0
         return VersionInfo(major, minor, micro, releaselevel, serial)
-    except Exception:
-        raise ImportError("Invalid package version {}".format(ver))
+    except Exception as e:
+        raise ImportError("Invalid package version {}".format(ver)) from e
```
Comment (webknjaz) [inline aiopg/__init__.py]: FTR, in Python 3, exceptions are chained automatically unless you're doing something weird...

**Comment by vir-mir** [comment]
in master
